UnitySDK/Assets/ML-Agents/Scripts/Academy.cs

```diff
@@ -138,10 +154,71 @@ public void LazyInitialization()
             }
         }
 
+        /// <summary>
+        /// Add Academy.Instance.EnvironmentStep to the FixedUpdate step of the Player Loop.
+        /// You can also call Academy.Instance.EnvironmentStep instead of connecting to the loop.
+        /// Note that if you've already made customizations to the Player Loop, this may overwrite those.
+        /// </summary>
+        public void ConnectToPlayerLoop()
+        {
+            if (m_ConnectedToPlayerLoop)
+            {
+                return;
+            }
+            m_ConnectedToPlayerLoop = ModifyGameLoop(true);
+        }
+
+        /// <summary>
+        /// Remove Academy.Instance.EnvironmentStep from the FixedUpdate step of the Player Loop.
+        /// Note that if you've already made customizations to the Player Loop, this may overwrite those.
+        /// </summary>
+        public void DisconnectFromPlayerLoop()
+        {
+            if (!m_ConnectedToPlayerLoop)
+            {
+                return;
+            }
+            var success = ModifyGameLoop(true);
+            m_ConnectedToPlayerLoop = !success;
+        }
+
+        /// <summary>
+        /// Add or remove the Academy step from the FixedUpdate step of the Player Loop
+        /// </summary>
+        /// <param name="connect">whether to connect or disconnect</param>
+        /// <returns>Whether the loop was modified.</returns>
+        bool ModifyGameLoop(bool connect)
+        {
+            var playerLoop = PlayerLoop.GetDefaultPlayerLoop();
+            for (var i=0; i< playerLoop.subSystemList.Length; i++)
+            {
+                var subSystem = playerLoop.subSystemList[i];
+                if (subSystem.type == typeof(FixedUpdate))
+                {
+                    if (true)
```

Comment:
is this just for scoping? You can remove the `if` and just leave the braces.

Reply:
Whoops, there use to be an "else" there when I was testing.